game/songs.cc

```diff
@@ -94,18 +95,25 @@ void Songs::reload_internal() {
 	Paths systemSongs = getPathsConfig("paths/system-songs");
 	Paths paths = getPathsConfig("paths/songs");
 	paths.insert(paths.begin(), systemSongs.begin(), systemSongs.end());
-
+	std::vector<std::future<void>> futures;
 	for (auto it = paths.begin(); m_loading && it != paths.end(); ++it) { //loop through stored directories from config
-		try {
-			if (!fs::is_directory(*it)) { std::clog << "songs/info: >>> Not scanning: " << *it << " (no such directory)\n"; continue; }
-			std::clog << "songs/info: >>> Scanning " << *it << std::endl;
-			size_t count = loadedSongs();
-			reload_internal(*it);
-			size_t diff = loadedSongs() - count;
-			if (diff > 0 && m_loading) std::clog << "songs/info: " << diff << " songs loaded\n";
-		} catch (std::exception& e) {
-			std::clog << "songs/error: >>> Error scanning " << *it << ": " << e.what() << '\n';
-		}
+		futures.push_back(std::async(std::launch::async, [this, it] {
+			try {
+				if (!fs::is_directory(*it)) { std::clog << "songs/info: >>> Not scanning: " << *it << " (no such directory)\n"; return; }
+				std::clog << "songs/info: >>> Scanning " << *it << std::endl;
+				size_t count = loadedSongs();
+				reload_internal(*it);
+				size_t diff = loadedSongs() - count;
+				if (diff > 0 && m_loading) std::clog << "songs/info: " << diff << " songs loaded\n";
```

Comment:
Would be good if here will be the path logged too.

Reply:
Done :D